core/src/avm2/verify.rs

```diff
@@ -0,0 +1,1193 @@
+                                previous_op = Some(op.clone());
+                                *op = Op::Nop;
+                                continue;
+                            }
+                        }
+                        _ => {}
+                    },
+                    Op::GetProperty { index: name_index } => match previous_op_some {
+                        Op::GetLocal { index: local_index } => {
+                            let class = local_types[local_index as usize];
+                            if let Some(class) = class {
+                                let multiname = method
+                                    .translation_unit()
+                                    .pool_maybe_uninitialized_multiname(
+                                        *name_index,
+                                        &mut activation.context,
+                                    );
+
+                                if let Ok(multiname) = multiname {
+                                    if !multiname.has_lazy_component() {
+                                        match class.instance_vtable().get_trait(&multiname) {
+                                            Some(Property::Slot { slot_id })
+                                            | Some(Property::ConstSlot { slot_id }) => {
+                                                previous_op = Some(op.clone());
+                                                *op = Op::GetSlot { index: slot_id };
+                                                continue;
+                                            }
+                                            Some(Property::Virtual { get: Some(get), .. }) => {
+                                                previous_op = Some(op.clone());
+                                                *op = Op::CallMethod {
+                                                    num_args: 0,
+                                                    index: Index::new(get),
+                                                };
+                                                continue;
+                                            }
+                                            _ => {}
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                        _ => {}
+                    },
+                    Op::AsType {
+                        type_name: name_index,
+                    } => {
+                        let multiname = method
+                            .translation_unit()
+                            .pool_maybe_uninitialized_multiname(
+                                *name_index,
+                                &mut activation.context,
+                            );
+
+                        let resolved_type = if let Ok(multiname) = multiname {
+                            if !multiname.has_lazy_component() {
```

Comment:
I think a failed/lazy multiname should throw a VerifyError at an earlier stage, right?
(can be a follow-up)